ast/flatbuffers_test.go

```diff
@@ -1,12 +1,120 @@
+// +build libflux
+
 package ast_test
 
 import (
+	"regexp"
 	"testing"
 
 	flatbuffers "github.com/google/flatbuffers/go"
+	"github.com/google/go-cmp/cmp"
+	"github.com/influxdata/flux/ast"
 	"github.com/influxdata/flux/ast/internal/fbast"
+	gparser "github.com/influxdata/flux/internal/parser"
+	"github.com/influxdata/flux/internal/token"
+	"github.com/influxdata/flux/libflux/go/libflux"
+	//rparser "github.com/influxdata/flux/parser"
 )
 
+var CompareOptions = []cmp.Option{
+	cmp.Transformer("", func(re *regexp.Regexp) string {
+		if re == nil {
+			return "<nil>"
+		}
+		return re.String()
+	}),
+}
+
+func TestRoundTrip(t *testing.T) {
+	srcs := [2]string{`
+package mypkg
+import "my_other_pkg"
+import "yet_another_pkg"	
+option now = () => (2030-01-01T00:00:00Z)
+option foo.bar = "baz"
+builtin foo
+
+# // bad stmt
+
+test aggregate_window_empty = () => ({
+    input: testing.loadStorage(csv: inData),
+    want: testing.loadMem(csv: outData),
+    fn: (table=<-) =>
+        table
+            |> range(start: 2018-05-22T19:53:26Z, stop: 2018-05-22T19:55:00Z)
+            |> aggregateWindow(every: 30s, fn: sum),
+})
+`, `
+a
+
+arr = [0, 1, 2]
+f = (i) => i
+ff = (i=<-, j) => {
+  k = i + j
+  return k
+}
+b = z and y
+b = z or y
+o = {red: "red", "blue": 30}
+m = o.red
+i = arr[0]
+n = 10 - 5 + 10
+n = 10 / 5 * 10
+m = 13 % 3
+p = 2^10
+b = 10 < 30
+b = 10 <= 30
+b = 10 > 30
+b = 10 >= 30
+eq = 10 == 10
+neq = 11 != 10
+b = not false
+e = exists o.red
+tables |> f()
+fncall = id(v: 20)
+fncall2 = foo(v: 20, w: "bar")
+v = if true then 70.0 else 140.0 
+ans = "the answer is ${v}"
+paren = (1)
+
+i = 1
+f = 1.0
+s = "foo"
+d = 10s
+b = true
+dt = 2030-01-01T00:00:00Z
+re =~ /foo/
+re !~ /foo/
+bad_expr = 3 * + 1
+// bad_expr = 3 * / 1
+`}
+	for _, src := range srcs {
+		astFbs := libflux.ParseIntoFbs(src)
+
+		srcb := []byte(src)
+		f := token.NewFile("", len(src))
+		file := gparser.ParseFile(f, srcb)
+		packageName := "main"
+		if file.Package != nil && file.Package.Name != nil {
+			packageName = file.Package.Name.Name
+		}
+		astGo := &ast.Package{
+			Package: packageName,
+			Files:   []*ast.File{file},
+		}
+		//astRust := rparser.ParseSource(src)
+
+		if !cmp.Equal(astFbs, astGo, CompareOptions...) {
+			t.Errorf("AST roundtrip vs. Go unexpected packages -fbs/+go:\n%s",
+				cmp.Diff(astFbs, astGo, CompareOptions...))
+		}
+		//if !cmp.Equal(astFbs, astRust, CompareOptions...) {
```

Comment:
Can you link to the related issues in GitHub here?

Reply:
Actually, if you can create a more liberal `CompareOptions` that would allow Rust-JSON AST comparison to succeed, it might be good to do that here, instead of commenting out code.  I'm okay with either way.